File: 7-kyu/chuck-norris-i---push-ups/python/solution.py
def chuck_push_ups(st):
    if type(st) != str or len(st) == 0:
        return 'FAIL!!'
    numbers = [''.join(ch for ch in elem if ch == '1' or ch == '0') for elem in st.split()]
    numbers = list(filter(lambda elem: len(elem) > 1, numbers))
    if len(numbers) < 1:
        return 'CHUCK SMASH!!'
    return max(int(elem, 2) for elem in numbers)

File: 7-kyu/chuck-norris-i---push-ups/javascript/solution.js
const chuckPushUps = st => {
  if (typeof st !== "string" || st.length === 0) {
    return "FAIL!!";
  }
  const numbers = st.split(" ")
    .map(elem => elem.split("").filter(ch => ch === "1" || ch === "0").join(""))
    .filter(elem => elem.length > 1);
  if (numbers.length === 0) {
    return "CHUCK SMASH!!";
  }
  return Math.max(...(numbers.map(n => parseInt(n, 2))));
};
